botorch/optim/initializers.py

```diff
@@ -63,6 +63,184 @@
+    if indices[:, 1].max() >= d:
+        raise ValueError(
+            f"Constraint indices cannot exceed the problem dimension {d=}."
+        )
+    return (
+        torch.tensor([r[0] * d + r[1] for r in indices], dtype=torch.int64),
+        coefficients,
+        rhs,
+    )
+
+
+def sample_q_batches_from_polytope(
+    n: int,
+    q: int,
+    bounds: Tensor,
+    n_burnin: int,
+    thinning: int,
+    seed: int,
+    inequality_constraints: Optional[List[Tuple[Tensor, Tensor, float]]] = None,
+    equality_constraints: Optional[List[Tuple[Tensor, Tensor, float]]] = None,
+) -> Tensor:
+    r"""Samples `n` q-baches from a polytope of dimension `d`.
+
+    Args:
+        n: Number of q-batches to sample.
+        q: Number of samples per q-batch
+        bounds: A `2 x d` tensor of lower and upper bounds for each column of `X`.
+        n_burnin: The number of burn-in samples for the Markov chain sampler.
+        thinning: The amount of thinning.
+        seed: The random seed.
+        inequality constraints: A list of tuples (indices, coefficients, rhs),
+            with each tuple encoding an inequality constraint of the form
+            `\sum_i (X[indices[i]] * coefficients[i]) >= rhs`.
+        equality constraints: A list of tuples (indices, coefficients, rhs),
+            with each tuple encoding an inequality constraint of the form
+            `\sum_i (X[indices[i]] * coefficients[i]) = rhs`.
+
+    Returns:
+        A `n x q x d`-dim tensor of samples.
+    """
+
+    # check if inter-point constraints are present
+    inter_point = False
+    if inequality_constraints is not None:
+        for c in inequality_constraints:
+            if len(c[0].shape) > 1:
+                inter_point = True
+                break
+    if inter_point is False and equality_constraints is not None:
+        for c in equality_constraints:
+            if len(c[0].shape) > 1:
+                inter_point = True
+                break
+
+    if inter_point:
+        return (
+            get_polytope_samples(
+                n=n,
+                bounds=torch.hstack([bounds for _ in range(q)]),
+                inequality_constraints=transform_constraints(
+                    constraints=inequality_constraints, q=q, d=bounds.shape[1]
+                ),
+                equality_constraints=transform_constraints(
+                    constraints=equality_constraints, q=q, d=bounds.shape[1]
+                ),
+                seed=seed,
+                n_burnin=n_burnin,
+                thinning=thinning * q,
+            )
+            .view(n, q, -1)
+            .cpu()
+        )
+    return (
+        get_polytope_samples(
+            n=n * q,
+            bounds=bounds,
+            inequality_constraints=inequality_constraints,
+            equality_constraints=equality_constraints,
+            seed=seed,
+            n_burnin=n_burnin,
+            thinning=thinning,
+        )
+        .view(n, q, -1)
+        .cpu()
+    )
```

Comment:
Again a stylistic / conciseness nit

```suggestion
    if inter_point:
        samples = get_polytope_samples(
            n=n,
            bounds=torch.hstack([bounds for _ in range(q)]),
            inequality_constraints=transform_constraints(
                constraints=inequality_constraints, q=q, d=bounds.shape[1]
            ),
            equality_constraints=transform_constraints(
                constraints=equality_constraints, q=q, d=bounds.shape[1]
            ),
            seed=seed,
            n_burnin=n_burnin,
            thinning=thinning * q,
        )
    else:
        samples = get_polytope_samples(
            n=n * q,
            bounds=bounds,
            inequality_constraints=inequality_constraints,
            equality_constraints=equality_constraints,
            seed=seed,
            n_burnin=n_burnin,
            thinning=thinning,
        )
    return samples.view(n, q, -1).cpu()
```